proto/message.py

```diff
@@ -662,7 +682,24 @@ def __getattr__(self, key):
             more details.
         """
         try:
-            pb_type = self._meta.fields[key].pb_type
+            try:
+                pb_type = self._meta.fields[key].pb_type
+            except KeyError:
+                # Underscores may be appended to field names
+                # that collide with python or proto-plus keywords.
+                # In case a key only exists with a `_` suffix, coerce the key
+                # to include the `_` suffix. It's not possible to
+                # natively define the same field with a trailing underscore in protobuf.
+                # See related issue
+                # https://github.com/googleapis/python-api-core/issues/227
+                if f"{key}_" in self._meta.fields:
+                    key = f"{key}_"
+                    pb_type = self._meta.fields[key].pb_type
+                else:
+                    raise KeyError(
+                        "Unknown field for {}: {}".format(self.__class__.__name__, key)
+                    )
```

Comment:
I think this should actually be an `AttributeError` for consistency with the exception below.